src/examples/algorithms/HelloSP_TP.cpp

```diff
@@ -0,0 +1,87 @@
+/* ---------------------------------------------------------------------
+ * Numenta Platform for Intelligent Computing (NuPIC)
+ * Copyright (C) 2013-2015, Numenta, Inc.  Unless you have an agreement
+ * with Numenta, Inc., for a separate license for this software code, the
+ * following terms and conditions apply:
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see http://www.gnu.org/licenses.
+ *
+ * http://numenta.org/licenses/
+ * ---------------------------------------------------------------------
+ */
+
+#include <iostream>
+#include <vector>
+#include <algorithm>    // std::generate
+#include <ctime>        // std::time
+#include <cstdlib>      // std::rand, std::srand
+#include <cmath> 	// pow
+
+#include "nupic/algorithms/SpatialPooler.hpp"
+#include "nupic/algorithms/Cells4.hpp"
+
+using namespace std;
+using namespace nupic;
+using namespace nupic::algorithms::spatial_pooler;
+using namespace nupic::algorithms::Cells4;
+
+// function generator:
+int RandomNumber01 () { return (rand()%2); } // returns random (binary) numbers from {0,1}
```

Comment:
@scottpurdy I'm having trouble using `nupic::Random` with the `generate()` funcion: 

```
>                  from /home/mmm/nupic/nupic.core/src/examples/algorithms/HelloSP_TP.cpp:25:
/usr/include/c++/4.8/bits/stl_algo.h: In instantiation of ‘void std::generate(_FIter, _FIter, _Generator) [with _FIter = __gnu_cxx::__normal_iterator<unsigned int*, std::vector<unsigned int> ; _Generator = unsigned int]’:
/home/mmm/nupic/nupic.core/src/examples/algorithms/HelloSP_TP.cpp:69:50:   required from here
/usr/include/c++/4.8/bits/stl_algo.h:5058:19: error: ‘__gen’ cannot be used as a function
  *__first = __gen();
```

when I do: 

```
nupic::Random rand01; 
generate(..., ..., rand01());
```